### PR TITLE
PulseOut working

### LINKED ISSUE #1044: nrf: implement pulseio.PulseOut

### PR DESCRIPTION
**Pull request by dhalbert**
Implement `PulseOut`. Fixes #1044. Tested with a Saleae.

Some cleanup on copyright notices and `PulseOut` doc.
Added general timer allocation/deallocation to `nrf/peripherals`.

Make `HardFault_Handler` loop, so we can get a stack trace.

### CODE REVIEW

File: ports/nrf/common-hal/pulseio/PulseOut.c [hunk 1]
```diff
@@ -29,31 +29,136 @@
...
 bool common_hal_pulseio_pulseout_deinited(pulseio_pulseout_obj_t* self) {
-  return 1;
+    return self->pwmout == NULL;
 }
 
 void common_hal_pulseio_pulseout_deinit(pulseio_pulseout_obj_t* self) {
+    if (common_hal_pulseio_pulseout_deinited(self)) {
+        return;
+    }
+    turn_on(self);
```
Comment (tannewt) [inline ports/nrf/common-hal/pulseio/PulseOut.c]:
```suggestion
    turn_off(self);
```

Should this turn off instead?

Comment (dhalbert) [reply ports/nrf/common-hal/pulseio/PulseOut.c]: This is the same behavior as in the `atmel-samd` port. When you deinit, it turns the carrier pin back on, as it was before you wrapped it in the `PulseOut()`.

Comment (dhalbert) [reply ports/nrf/common-hal/pulseio/PulseOut.c]: https://github.com/adafruit/circuitpython/blob/master/ports/atmel-samd/common-hal/pulseio/PulseOut.c#L166

Comment (tannewt) [reply ports/nrf/common-hal/pulseio/PulseOut.c]: Ah ok! Thanks for the reminder.

**Pull request merged**